File: app/src/main/java/com/example/aichat/ConfiguredModelPicker.java
package com.example.aichat;

import android.content.Context;

import java.util.ArrayList;
import java.util.List;

/**
 * 已配置的模型：厂商已设置 API Key（或本地厂商），且用户已添加该模型。
 * 用于模型配置选择器等。
 */
public class ConfiguredModelPicker {

    public static class Option {
        public String providerId;
        public String providerName;
        public String modelId;
        public String displayName;  // nickname or modelId

        /** 存储格式，用于 ConfigManager */
        public String getStorageKey() {
            return providerId + ":" + modelId;
        }

        public static Option fromStorageKey(String key, Context ctx) {
            if (key == null || !key.contains(":")) return null;
            int i = key.indexOf(':');
            String pid = key.substring(0, i);
            String mid = key.substring(i + 1);
            ProviderManager pm = new ProviderManager(ctx);
            ProviderInfo p = pm.getProvider(pid);
            if (p == null) return null;
            if (p.models == null) return null;
            for (ProviderInfo.ProviderModelInfo m : p.models) {
                if (m != null && mid.equals(m.modelId)) {
                    Option o = new Option();
                    o.providerId = pid;
                    o.providerName = p.name;
                    o.modelId = mid;
                    o.displayName = m.nickname != null && !m.nickname.isEmpty() ? m.nickname : mid;
                    return o;
                }
            }
            Option o = new Option();
            o.providerId = pid;
            o.providerName = p.name;
            o.modelId = mid;
            o.displayName = mid;
            return o;
        }
    }

    /** 获取所有可选的已配置模型 */
    public static List<Option> getConfiguredModels(Context ctx) {
        ProviderManager pm = new ProviderManager(ctx);
        List<ProviderInfo> providers = pm.getAllProviders();
        List<Option> result = new ArrayList<>();
        ProviderCatalog.CatalogItem cat;
        for (ProviderInfo p : providers) {
            cat = ProviderCatalog.get(p.id);
            boolean needsKey = cat == null || cat.needsKey;
            boolean hasHost = p.apiHost != null && !p.apiHost.isEmpty();
            boolean hasKey = !needsKey || (p.apiKey != null && !p.apiKey.isEmpty());
            if (!hasHost || !hasKey) continue;
            if (p.models == null || p.models.isEmpty()) continue;
            for (ProviderInfo.ProviderModelInfo m : p.models) {
                if (m.modelId == null) continue;
                Option o = new Option();
                o.providerId = p.id;
                o.providerName = p.name;
                o.modelId = m.modelId;
                o.displayName = m.nickname != null && !m.nickname.isEmpty() ? m.nickname : m.modelId;
                result.add(o);
            }
        }
        return result;
    }
}

File: app/src/main/java/com/example/aichat/ConfiguredModelPicker.kt
package com.example.aichat

import android.content.Context

/**
 * 已配置的模型：厂商已设置 API Key（或本地厂商），且用户已添加该模型。
 * 用于模型配置选择器等。
 */
class ConfiguredModelPicker {

    class Option {
        @JvmField var providerId: String? = null
        @JvmField var providerName: String? = null
        @JvmField var modelId: String? = null
        @JvmField var displayName: String? = null  // nickname or modelId

        /** 存储格式，用于 ConfigManager */
        fun getStorageKey(): String = "$providerId:$modelId"

        companion object {
            @JvmStatic
            fun fromStorageKey(key: String?, ctx: Context): Option? {
                if (key == null || !key.contains(":")) return null
                val i = key.indexOf(':')
                val pid = key.substring(0, i)
                val mid = key.substring(i + 1)
                val pm = ProviderManager(ctx)
                val p = pm.getProvider(pid) ?: return null
                if (p.models == null) return null
                for (m in p.models) {
                    if (m != null && mid == m.modelId) {
                        val o = Option()
                        o.providerId = pid
                        o.providerName = p.name
                        o.modelId = mid
                        o.displayName = if (!m.nickname.isNullOrEmpty()) m.nickname else mid
                        return o
                    }
                }
                val o = Option()
                o.providerId = pid
                o.providerName = p.name
                o.modelId = mid
                o.displayName = mid
                return o
            }
        }
    }

    companion object {
        /** 获取所有可选的已配置模型 */
        @JvmStatic
        fun getConfiguredModels(ctx: Context): List<Option> {
            val pm = ProviderManager(ctx)
            val providers = pm.getAllProviders()
            val result = ArrayList<Option>()
            for (p in providers) {
                val cat = ProviderCatalog.get(p.id)
                val needsKey = cat == null || cat.needsKey
                val hasHost = !p.apiHost.isNullOrEmpty()
                val hasKey = !needsKey || !p.apiKey.isNullOrEmpty()
                if (!hasHost || !hasKey) continue
                if (p.models == null || p.models.isEmpty()) continue
                for (m in p.models) {
                    if (m.modelId == null) continue
                    val o = Option()
                    o.providerId = p.id
                    o.providerName = p.name
                    o.modelId = m.modelId
                    o.displayName = if (!m.nickname.isNullOrEmpty()) m.nickname else m.modelId
                    result.add(o)
                }
            }
            return result
        }
    }
}
